### DISCUSSION TITLE
[Bug]: Pimcore not installing, database credentials not stored

### DISCUSSION THREAD

**Discussion by mick340** [pimcore/pimcore#10751] [Q&A / Support]
### Expected behavior

When running pimcore install i expect to install pimcore

### Actual behavior

Pimocre is not installing because of Step 6/11 database setup

### Steps to reproduce

I have created a `acceptatie` user to connect to database `pimcore` @ `localhost`

As you below i can connect with the database using tableplus (connection trough ssh)

![Screenshot 2021-11-02 at 16 01 54](https://user-images.githubusercontent.com/8590308/139873118-2f9fd775-7a96-4f02-9676-d136e223a355.png)

When installing pimcore im using the optional parameters to give the required database information (Even the config/installer.yaml file isn't working)

![Screenshot 2021-11-02 at 16 03 52](https://user-images.githubusercontent.com/8590308/139873854-57bcce17-e589-4534-be9b-fef5d3197b9b.png)

I fill in the Admin username and password, and the installer starts and fails.

![Screenshot 2021-11-02 at 16 04 15](https://user-images.githubusercontent.com/8590308/139874252-ce0b6a41-0eb4-40c2-bab6-ed0310461be4.png)

The strange part is that it gives me an error that root@localhost is denied and that is true, because i don't want Pimcore to connect with the root user. It looks like that the given database credentials are not stored / taken over.

But when i try to install pimcore with wrong credentials (wrong password) it stops installing at step 0/11

So somewhere along the installing steps the database credentials are gone.

![Screenshot 2021-11-02 at 16 11 26](https://user-images.githubusercontent.com/8590308/139874873-77ccdf6a-b50b-4989-bae4-44b97d9baace.png)

**Comment by brusch** [comment]
I think you have already a database-config in place that causes the issue. 
The option `--ignore-existing-config` is just not overriding existing config, but doesn't mean it's not loading it.

**Reply by mick340** [reply]
Hello, where can i find this existing database config? I deleted the `config/local/database.yaml` file. I also installed the project from 0 so there should be no config stored. 

After i ran the installer command there is a new database.yaml file which has the correct username, password and database name.

**Comment by mick340** [comment]
Could you help me with this? Im stuck and don't know how to go forward from this

**Comment by dvesh3** [comment]
@mick340 Did you try with `--skip-database-config` option?

**Reply by mick340** [reply]
Yes, i have just tried it with the `--skip-database-config` and i still get the same error 

` * An exception occurred in driver: SQLSTATE[HY000] [1045] Access denied for user 'root'@'localhost' (using password: NO)`

**Comment by mick340** [comment]
@dvesh3 or @brusch I'm still stuck, we cannot install Pimcore because of the database error. Could you guys help me with this issue?

**Reply by damian1424** [reply]
I had a similar problem, try this way:
`export PIMCORE_INSTALL_MYSQL_HOST_SOCKET=localhost:3306; vendor/bin/pimcore-install`


If you want, you can enter the user and password of the database right away, otherwise you have the option to enter them in the console. 

If that doesn't work, also check what's in the var/log. Maybe there will be a logical explanation of the problem.

**Reply by mick340** [reply]
> I had a similar problem, try this way: `export PIMCORE_INSTALL_MYSQL_HOST_SOCKET=localhost:3306; vendor/bin/pimcore-install`
> 
> If you want, you can enter the user and password of the database right away, otherwise you have the option to enter them in the console.
> 
> If that doesn't work, also check what's in the var/log. Maybe there will be a logical explanation of the problem.

I have already tried this, it isn't working sadly :(

**Reply by damian1424** [reply]
Does your server fit all system requirements?

https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Requirements.html

**Reply by dvesh3** [reply]
@mick340 please check if you are using correct installer config, so if using Pimcore version < 10.1 then use `app/config/installer.yml` or for newer version use `config/installer.yaml` with `--skip-database-config` option.

https://github.com/pimcore/pimcore/commit/da1c620994e461d79e91a1e8495e722e6154a32f#diff-8e789ff505cab9d3406d3f98e6856e6a0d42dfeeb5c4ee14d7f691c00df1d3c0

**Reply by mick340** [reply]
@dvesh3 Im using Pimcore 10.2.3 and my installer.yaml is located in the `config` directory. But it still asks me for credentials when i try to install pimore

**Reply by Anaxarchos** [reply]
Thanks, This worked for me, as it was the easiest way to pass the Port to the pimcore installer

**Comment by mick340** [comment]
Is there any other solution possible?

**Comment by alexej-d** [comment]
I have exactly the same problem with Pimcore 10.2.2 :(

**Reply by alexej-d** [reply]
BTW: APP_ENV=dev ./vendor/bin/pimcore-install works. Using prod environment curiously sets the database credentials to doctrines defaults when calling `\Pimcore\Db::get()` in Installer.php…

**Reply by alexej-d** [reply]
Ignore my comments. I have made the mistake of calling the default APP_ENV `production` instead of `prod`
